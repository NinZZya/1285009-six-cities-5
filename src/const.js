export const AppPath = {
  ROOT: `/`,
  LOGIN: `/login`,
  CITY: `/city`,
  OFFER: `/offer`,
  FAVORITES: `/favorites`,
  NOT_FOUND: `/not-found`,
};

export const CITIES = {
  1: {id: 1, name: `Paris`},
  2: {id: 2, name: `Cologne`},
  3: {id: 3, name: `Brussels`},
  4: {id: 4, name: `Amsterdam`},
  5: {id: 5, name: `Hamburg`},
  6: {id: 6, name: `Dusseldorf`},
};

export const SortType = {
  POPULAR: `Popular`,
  LOW_TO_HIGH: `Price: low to high`,
  HIGH_TO_LOW: `Price: high to low`,
  TOP_RATED_FIRST: `Top rated first`,
};

export const STARS = [
  {value: 5, title: `perfect`},
  {value: 4, title: `good`},
  {value: 3, title: `not bad`},
  {value: 2, title: `badly`},
  {value: 1, title: `terribly`},
];
