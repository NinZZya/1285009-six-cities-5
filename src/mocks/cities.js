export default {
  1: {
    id: 1,
    name: `Paris`,
    coords: [48.864716, 2.349014],
  },
  2: {
    id: 2,
    name: `Cologne`,
    coords: [50.9415491, 6.9408247],
  },
  3: {
    id: 3,
    name:
    `Brussels`,
    coords: [50.8503396, 4.3517103],
  },
  4: {
    id: 4,
    name: `Amsterdam`,
    coords: [52.38333, 4.9],
  },
  5: {id: 5, name: `Hamburg`},
  6: {id: 6, name: `Dusseldorf`},
};
