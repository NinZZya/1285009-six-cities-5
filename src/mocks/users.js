export default [{
  id: 0,
  name: `test`,
  email: `test@test.test`,
  password: `test`,
}];
