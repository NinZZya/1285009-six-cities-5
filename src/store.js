import {createStore, applyMiddleware} from 'redux';
import thunk from 'redux-thunk';
import {composeWithDevTools} from 'redux-devtools-extension';
import rootReduser from './redux/root-reducer';
import Api from './api/api';


export default createStore(
    rootReduser,
    composeWithDevTools(
        applyMiddleware(thunk.withExtraArgument(Api))
    )
);
