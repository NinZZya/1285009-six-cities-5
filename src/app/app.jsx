import React from 'react';
import {useLocation} from 'react-router-dom';
import {connect} from 'react-redux';
import AppRoute from '../routes/app-route/app-route';
import {getUser, getUserStatus} from '../reducer/user/user-selectors';
import * as Type from '../types';
import {AppPath} from '../const';


const PathToClassName = {
  [AppPath.ROOT]: `page--gray page--main`,
  [AppPath.CITY]: `page--gray page--main`,
  [AppPath.LOGIN]: `page--gray page--login`,
  [AppPath.FAVORITES]: ``,
  [AppPath.OFFER]: ``,
  [AppPath.NOT_FOUND]: `page--gray page--main`,
};

const getPage = (path) => {
  const pathArgs = path.split(`/`);

  return pathArgs.length === 1 ? `/` : `/${pathArgs[1]}`;
};

const App = (props) => {
  const {userStatus, user} = props;
  const location = useLocation();
  const page = getPage(location.pathname);

  return (
    <div
      className={`page ${PathToClassName[page]}`}
    >
      <AppRoute
        userStatus={userStatus}
        user={user}
      />
    </div>
  );
};

App.propTypes = {
  userStatus: Type.USER_STATUS,
  user: Type.USER,
};

const mapStateToProps = (state) => ({
  userStatus: getUserStatus(state),
  user: getUser(state),
});

export {App};
export default connect(mapStateToProps)(App);
