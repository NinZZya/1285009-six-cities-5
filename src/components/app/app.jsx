import React from 'react';
import {Switch, Route, Redirect, useLocation} from 'react-router-dom';
import Header from '../header/header';
import Footer from '../footer/footer';
import Main from '../../pages/main/main';
import Login from '../../pages/login/login';
import Offer from '../../pages/offer/offer';
import Favorites from '../../pages/favorites/favorites';
import {AppRoute} from '../../const';
import * as Type from '../../types';


const LocationToClassName = {
  [AppRoute.ROOT]: `page--gray page--main`,
  [AppRoute.LOGIN]: `page--gray page--login`,
  [AppRoute.FAVORITES]: ``,
  [AppRoute.OFFER]: ``,
};

const getPage = (path) => {
  const pathArgs = path.split(`/`);

  return pathArgs.length > 2 ? `/${pathArgs[1]}/` : `/`;
};

const App = ({activeCityId}) => {
  const location = useLocation();
  const page = getPage(location.pathname);

  return (
    <div
      className={`page ${LocationToClassName[page]}`}
    >
      <Header />
      <Switch>
        <Route exact path={AppRoute.ROOT}>
          <Main activeCityId={activeCityId} />
        </Route>
        <Route exact path={AppRoute.LOGIN} component={Login} />
        <Route exact path={`${AppRoute.OFFER}:id?`} component={Offer} />
        <Route exact path={AppRoute.FAVORITES}>
          <Favorites />
          <Footer />
        </Route>
        <Redirect to={AppRoute.ROOT} />
      </Switch>
    </div>
  );
};

App.propTypes = {
  activeCityId: Type.id,
};

export default App;
