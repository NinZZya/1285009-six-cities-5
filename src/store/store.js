import {createStore, applyMiddleware} from 'redux';
import thunk from 'redux-thunk';
import {composeWithDevTools} from 'redux-devtools-extension';
import rootReduser from './reducer/reducer';
import {changeUserStatus} from './reducer/user/user-actions';
import {UserStatus} from '../constants/const';
import Api from '../services/api';


const onUnauthorized = () => store.dispatch(changeUserStatus(UserStatus.NO_AUTH));

const api = new Api({
  onFailCallback: onUnauthorized,
});

const store = createStore(
    rootReduser,
    composeWithDevTools(
        applyMiddleware(thunk.withExtraArgument(api))
    )
);


export default store;
