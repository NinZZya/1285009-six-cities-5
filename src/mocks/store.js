import NameSpace from '../store/reducer/name-space';
import offers, {offersMap} from './offers';
import cities from './cities';
import reviews from './reviews';
import {user} from './users';
import {SortType, DataStatus, UserStatus} from '../constants/const';

const ACTIVE_CITY_ID = 4;
const ERROR = `ERROR`;
const DATA_STATUS = DataStatus.SUCCESS;
const SORT_TYPE = SortType.HIGH_TO_LOW;

const citiesState = {
  cities,
  activeId: ACTIVE_CITY_ID,
};

const offersState = {
  status: DATA_STATUS,
  offers: offersMap,
  favorites: offersMap,
  sortType: SORT_TYPE,
  offerStatus: DATA_STATUS,
  reviewsStatus: DATA_STATUS,
  reviews,
  nearOffersStatus: DATA_STATUS,
  nearOffers: offers,
};

const userState = {
  status: UserStatus.NO_AUTH,
  user,
  error: ERROR,
};

export default {
  [NameSpace.CITIES]: citiesState,
  [NameSpace.OFFERS]: offersState,
  [NameSpace.USER]: userState,
};
