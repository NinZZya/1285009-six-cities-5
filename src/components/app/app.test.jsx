import React from 'react';
import renderer from 'react-test-renderer';
import {Provider} from 'react-redux';
import {Router} from 'react-router-dom';
import {App} from './app';
import {UserStatus} from '../../constants/const';
import history from '../../constants/history';
import store from '../../store/store';
import {user} from '../../mocks/users';


describe(`Should App render correctly`, () => {
  it(`Render App`, () => {
    const tree = renderer
      .create(
          <Provider store={store}>
            <Router history={history}>
              <App
                userStatus={UserStatus.NO_AUTH}
                user={user}
              />
            </Router>
          </Provider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
