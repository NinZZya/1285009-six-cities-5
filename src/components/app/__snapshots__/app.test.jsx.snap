// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should App render correctly Render App 1`] = `
<div
  className="page page--gray page--main"
>
  <header
    className="header"
  >
    <div
      className="container"
    >
      <div
        className="header__wrapper"
      >
        <div
          className="header__left"
        >
          <a
            className="header__logo-link header__logo-link--active"
            href="/"
            onClick={[Function]}
          >
            <img
              alt="6 cities logo"
              className="header__logo"
              height="41"
              src="/img/logo.svg"
              width="81"
            />
          </a>
        </div>
        <nav
          className="header__nav"
        >
          <ul
            className="header__nav-list"
          >
            <li
              className="header__nav-item user"
            >
              <a
                className="header__nav-link header__nav-link--profile"
                href="/login"
                onClick={[Function]}
              >
                <div
                  className="header__avatar-wrapper user__avatar-wrapper"
                />
                <span
                  className="header__user-name user__name"
                >
                  Sign in
                </span>
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>
  <main
    className="page__main page__main--index page__main--index-empty"
  >
    <h1
      className="visually-hidden"
    >
      Cities
    </h1>
    <div
      className="tabs"
    >
      <section
        className="locations container"
      >
        <ul
          className="locations__list tabs__list"
        >
          <li
            className="locations__item"
          >
            <a
              className="locations__item-link tabs__item tabs__item--active"
              href="/city/1"
              onClick={[Function]}
            >
              <span>
                Paris
              </span>
            </a>
          </li>
          <li
            className="locations__item"
          >
            <a
              className="locations__item-link tabs__item "
              href="/city/2"
              onClick={[Function]}
            >
              <span>
                Cologne
              </span>
            </a>
          </li>
          <li
            className="locations__item"
          >
            <a
              className="locations__item-link tabs__item "
              href="/city/3"
              onClick={[Function]}
            >
              <span>
                Brussels
              </span>
            </a>
          </li>
          <li
            className="locations__item"
          >
            <a
              className="locations__item-link tabs__item "
              href="/city/4"
              onClick={[Function]}
            >
              <span>
                Amsterdam
              </span>
            </a>
          </li>
          <li
            className="locations__item"
          >
            <a
              className="locations__item-link tabs__item "
              href="/city/5"
              onClick={[Function]}
            >
              <span>
                Hamburg
              </span>
            </a>
          </li>
          <li
            className="locations__item"
          >
            <a
              className="locations__item-link tabs__item "
              href="/city/6"
              onClick={[Function]}
            >
              <span>
                Dusseldorf
              </span>
            </a>
          </li>
        </ul>
      </section>
    </div>
    <div
      className="cities"
    >
      <div
        className="container cities__places-container cities__places-container--empty"
      >
        <div
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
              "flexDirection": "column",
              "fontSize": "38px",
              "fontStyle": "oblique",
              "fontWeight": 700,
              "justifyContent": "flex-start",
              "lineHeight": 0.7,
              "margin": "20px",
              "padding": "30px",
              "textAlign": "center",
              "width": "100%",
            }
          }
        >
          <h3
            style={
              Object {
                "alignItems": "center",
                "display": "flex",
                "justifyContent": "space-between",
              }
            }
          >
            <img
              alt="logo"
              height="50"
              src="/img/logo.svg"
              style={
                Object {
                  "marginRight": "20px",
                }
              }
              width="100"
            />
            <svg
              height="100%"
              version="1.1"
              width="100%"
              xmlns="http://www.w3.org/2000/svg"
              xmlnsXlink="http://www.w3.org/1999/xlink"
            >
              <path
                id="path"
              >
                <animate
                  attributeName="d"
                  begin="0s"
                  dur="3s"
                  from="m0,110 h0"
                  repeatCount="indefinite"
                  to="m0,110 h1100"
                />
              </path>
              <text
                fill="black"
              >
                <textPath
                  xlinkHref="#path"
                >
                  Loading...
                </textPath>
              </text>
            </svg>
          </h3>
        </div>
      </div>
    </div>
  </main>
</div>
`;
